app/api/endpoints.py

```diff
@@ -39,7 +39,12 @@ def post_config(name: str):
     content = flask.request.get_json()
     nginx_path = flask.current_app.config['NGINX_PATH']
 
-    with io.open(os.path.join(nginx_path, name), 'w') as f:
+    config_file = os.path.join(nginx_path, name)
+
+    if not os.path.commonprefix(os.path.realpath(config_file), nginx_path):
+        return flask.make_response({'success': False}), 200
```

Comment:
I would choose a different status code. For example a 400 (Bad Request) or 403 (Forbidden).

Reply:
Also not that `os.commonprefix("/a/b", "/c/d")` returns `"/"` which is truthy, so the condition is not met.